tests/integration/queryrules.ts

```diff
@@ -0,0 +1,63 @@
+import { createMockedAnswersHeadless } from '../mocks/createMockedAnswersHeadless';
+
+const initialState = {
+  queryRules: {
+    actions: [
+      {
+        key: 'oldInfo',
+        data: {
+          someData: 'blah'
+        }
+      }
+    ]
+  }
+};
+
+const actions = [
+  {
+    key: 'trackingInfo',
+    data: {
+      apiUrl: 'www.example.com'
+    }
+  }
+];
+
+function mockSearchWithQueryRules() {
+  return Promise.resolve({
+    queryRulesActionsData: actions
+  });
+}
+
+describe('AnswersHeadless queryRules interactions properly update state', () => {
+  it('executeVerticalQuery properly updates queryRules state', async () => {
+    const answers = createMockedAnswersHeadless({
+      verticalSearch: mockSearchWithQueryRules
+    }, initialState);
+    answers.setVertical('123');
+    await answers.executeVerticalQuery();
+    const expectedState = {
+      queryRules: {
+        ...initialState.queryRules,
```

Comment:
Is this `...initialState.queryRules` line here and in the other test doing anything? It looks like it's just getting overridden by the actions below it

Reply:
Yeah, that was leftover from something else I was trying. I'll remove it!